### PR TITLE
Update FTB server

### LINKED ISSUE #1731: [Bug]: Installation for FTB Modpacks using Java 17 and the FTB egg not completing install
### Panel Version

1.9.1

### Wings Version

1.6.4

### Service

minecraft/java/ftb

### Modified

No, I did not modify the egg

### Expected Behavior

Install server, push start, eula popup, start server.

### Actual Behavior

Installing any of the 1.18 or 1.19 servers using java 17 and the ftb egg is not working.

During install most things download:
Patching com/mojang/blaze3d/vertex/VertexFormatElement$Usage$SetupState 1/1
The server installed successfully
You can delete this installer file now if you wish

the final configuration does not seem to complete
2022/06/30 10:19:38 Extracting jdk-17.0.2+8-jre/legal/java.xml/bcel.md -> jre/jdk-17.0.2+8-jre/legal/java.xml/bcel.md
2022/06/30 10:19:38 Installed!
mv: cannot stat 'forge-*.jar': No such file or directory

tried all 4 current servers using java 17
FTB Presents Direwolf20 1.18
FTB Plexiglass Mountain
FTB One
FTB Chocolate

all 4 stopped at "mv: cannot stat 'forge-*.jar': No such file or directory"

all expected mods and files have been downloaded.

The forge-server.jar does not complete

Tried a couple of 1.16 modpacks and the egg worked as expected

### Steps To Reproduce

Step 1: install latest egg-ftb-modpacksch-server.json
    "meta": {
        "version": "PTDL_v1",
        "update_url": null
    },
    "exported_at": "2021-06-14T21:28:54+03:00",
Step 2: Configure server to use Nest:Minecraft, Egg:FTB=modpacks.ch Server, Select java 17 yolk  Select FTB modpack ID i.e. 95 FTB Pack Version ID i.e. 2141 (example is for dirwold20 1.18 1.5.0  
Step 3 Save Modifications
Step 4 Reinstall Server

### Install logs

https://pteropaste.com/ok7z8n

### PR DESCRIPTION
**Pull request by manyrandomthings**
# Description

* Adds support for arm architecture
* Adds support for fabric modpacks
* Fixes forge 1.17+ modpacks (resolves #1731)
* Includes the given Log4jPatcher from the installer in the startup command

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

* [x] You verify that the start command applied does not use a shell script
  * [ ] If some script is needed then it is part of a current yolk or a PR to add one
* [x] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json [hunk 1]
```diff
@@ -13,23 +13,23 @@
...
...ER_TYPE=$([ \"$(uname -m)\" == \"x86_64\" ] && echo \"linux\" || echo \"arm\/linux\")\r\n\r\necho \"ModpackID: ${FTB_MODPACK_ID} VersionID: ${FTB_MODPACK_VERSION_ID} InstallerType: ${INSTALLER_TYPE}\"\r\n\r\n# download installer\r\ncurl -L https:\/\/api.modpacks.ch\/public\/modpack\/0\/0\/server\/${INSTALLER_TYPE} --output serversetup\r\nchmod +x .\/serversetup\r\n\r\n\r\n# remove old forge files (to allow updating)\r\nrm -rf libraries\/net\/minecraftforge\/forge\r\nrm -f unix_args.txt\r\n\r\n# run installer\r\n.\/serversetup ${FTB_MODPACK_ID} ${FTB_MODPACK_VERSION_ID} --auto --noscript --nojava\r\n\r\n# create symlink for forge unix_args.txt if exists\r\nif compgen -G \"libraries\/net\/minecraftforge\/forge\/*\/unix_args.txt\"; then\r\n    ln -sf libraries\/net\/minecraftforge\/forge\/*\/unix_args.txt unix_args.txt\r\nfi\r\n\r\n# move forge\/fabric jar file to start-server.jar if exists\r\nif compgen -G \"forge-*.jar\"; then\r\n    mv -f forge-*.jar start-server.jar\r\nelif compgen -G \"fabric-*.jar\"; then\r\n    mv -f fabric-*.jar start-server.jar\r\nfi\r\n\r\n# installer cleanup\r\nrm serversetup\r\nrm -f run.bat\r\nrm -f run.sh\r\n\r\necho \"Finished installing FTB modpack\"",
```
Comment (QuintenQVD0) [inline game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json:32]: I would prefer to keep this in functions as it was easy readble and please keep this:

```bash
if [ ! -d /mnt/server ]; then
 mkdir -p /mnt/server
fi
```

File: game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json [hunk 2]
```diff
@@ -13,23 +13,23 @@
...
-    ],
+    "docker_images": {
+        "Java 17": "ghcr.io\/pterodactyl\/yolks:java_17",
+        "Java 16": "ghcr.io\/pterodactyl\/yolks:java_16",
+        "Java 11": "ghcr.io\/pterodactyl\/yolks:java_11",
+        "Java 8": "ghcr.io\/pterodactyl\/yolks:java_8"
+    },
     "file_denylist": [],
-    "startup": "java -Xms128M -Xmx{{SERVER_MEMORY}}M -jar forge-server.jar",
+    "startup": "java -javaagent:log4jfix\/Log4jPatcher-1.0.0.jar -Xms128M -XX:MaxRAMPercentage=95.0 -Dterminal.jline=false -Dterminal.ansi=true $( [[  ! -f unix_args.txt ]] && printf %s \"-jar start-server.jar\" || printf %s \"@unix_args.txt\" )",
```
Comment (QuintenQVD0) [inline game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json:23]: does this use unix_args? and is the log4jfix/Log4jPatcher-1.0.0.jar file always present with that version?( maybe use a wildvard for the version)

File: game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json [hunk 3]
```diff
@@ -69,7 +72,8 @@
             "default_value": "",
             "user_viewable": true,
             "user_editable": true,
-            "rules": "nullable|string|max:20"
+            "rules": "nullable|integer|min:1",
```
Comment (QuintenQVD0) [inline game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json:75]: you can not have it nullable and have a min value. then set it to required and specify a max value

File: game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json [hunk 4]
```diff
@@ -51,7 +52,8 @@
...
```
Comment (QuintenQVD0) [inline game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json:55]: same here

File: game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json [hunk 5]
```diff
@@ -42,7 +42,8 @@
             "default_value": "",
             "user_viewable": true,
             "user_editable": true,
-            "rules": "nullable|string|max:99|min:4"
+            "rules": "nullable|string|max:99|min:4",
```
Comment (QuintenQVD0) [inline game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json:45]: you can not have nullable and a min value

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json (hunk 2 above)
Comment (manyrandomthings) [reply game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json:23]: Yes it launches with unix_args.txt, unless that doesn't exist, then it launches with start-server.jar (for either fabric or pre-1.17 forge)

And the installer always downloads version 1.0.0 of Log4jPatcher, from here: https://github.com/CreeperHost/modpacksch-serverdownloader/blob/fba9792cf71889ce973a6a0846a194e3113e1678/main.go#L405
However trying to use a wildcard for the version prevents the server from starting:
```
container@pterodactyl~ java -javaagent:log4jfix/Log4jPatcher-*.jar -Xms128M -XX:MaxRAMPercentage=95.0 -Dterminal.jline=false -Dterminal.ansi=true @unix_args.txt
Error opening zip file or JAR manifest missing : log4jfix/Log4jPatcher-*.jar
Error occurred during initialization of VM
agent library failed to init: instrument
container@pterodactyl~ Server marked as offline...
[Pterodactyl Daemon]: ---------- Detected server process in a crashed state! ----------
[Pterodactyl Daemon]: Exit code: 1
[Pterodactyl Daemon]: Out of memory: false
[Pterodactyl Daemon]: Aborting automatic restart, last crash occurred less than 60 seconds ago.
```

**Comment by manyrandomthings** [comment]
@QuintenQVD0 updated with your suggestions

**Comment by QuintenQVD0** [comment]
> @QuintenQVD0 updated with your suggestions

Thank you! I check as soon as I can

File: game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json (hunk 2 above)
Comment (QuintenQVD0) [reply game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json:23]: please try `-javaagent:log4jfix/Log4jPatcher-*.*.*.jar`

Comment (manyrandomthings) [reply game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json:23]: That doesn't work either. Should I just edit the script to add `mv log4jfix/Log4jPatcher-*.jar log4jfix/Log4jPatcher.jar` and change the startup to `-javaagent:log4jfix/Log4jPatcher.jar`, or should it just be left as 1.0.0?

Comment (QuintenQVD0) [reply game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json:23]: Judt leave it like this. It is oke. Then we will judt update the egg when they change it.

File: game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json (hunk 1 above)
Comment (Software-Noob) [reply game_eggs/minecraft/java/ftb/egg-ftb-modpacksch-server.json]: The whole IF statement is useless, as -p will already only create parent directories if necessary. This by itself is also not required, but parker runs the bash code locally.

**Comment by CubE135** [comment]
> > @QuintenQVD0 updated with your suggestions
> 
> Thank you! I check as soon as I can

Any updates on this?
Would love to use this egg for my FTB Server :)
Currently wokring with some janky forge egg solution :P

**Review by QuintenQVD0 — APPROVED** [review]
LGTM! thank you. 
Tested and working!

**Comment by CubE135** [comment]
Works perfectly, thank you very much :)

**Pull request merged**